os/arch/arm/src/armv7-m/up_blocktask.c

```diff
@@ -169,6 +169,9 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 			/* Restore the MPU registers in case we are switching to an application task */
 #ifdef CONFIG_ARMV7M_MPU
 			up_set_mpu_app_configuration(rtcb);
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+			up_set_mpu_stack_guard(STACKGUARD_MPU_NUM, rtcb);
```

Comment:
1. This API is dedicated to set MPU region for stack. Do we need to give `STACKGUARD_MPU_NUM` even it is fixed in header?
2. In SAM (Software Architecture Maturity), it checks duplication of codes. This is almost the same as `up_set_mpu_app_configuration`. Do we need to split them?

Reply:
1. Yes removed passing region number to the function
2. up_set_mpu_app_configuration function configuring 3 regions , need to change this function to use as general. 
Once we remove hard coded MPU numbers, we can use common function